src/uu/uuidgen/src/uuidgen.rs

```diff
@@ -0,0 +1,212 @@
+// This file is part of the uutils util-linux package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+#[cfg(target_os = "windows")]
+use windows::Win32::{
+    Foundation::{ERROR_BUFFER_OVERFLOW, ERROR_SUCCESS},
+    NetworkManagement::IpHelper::{
+        GetAdaptersAddresses, GAA_FLAG_SKIP_ANYCAST, GAA_FLAG_SKIP_DNS_SERVER,
+        GAA_FLAG_SKIP_FRIENDLY_NAME, GAA_FLAG_SKIP_MULTICAST, GAA_FLAG_SKIP_UNICAST,
+        IP_ADAPTER_ADDRESSES_LH,
+    },
+    Networking::WinSock::AF_UNSPEC,
+};
+
+use clap::{crate_version, Arg, ArgAction, Command};
+
+#[cfg(all(target_family = "unix", not(target_os = "redox")))]
+use nix::ifaddrs::getifaddrs;
+use uucore::{
+    error::{UResult, USimpleError},
+    format_usage, help_about, help_usage,
+};
+use uuid::Uuid;
+
+const ABOUT: &str = help_about!("uuidgen.md");
+const USAGE: &str = help_usage!("uuidgen.md");
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let args = uu_app().try_get_matches_from_mut(args)?;
+
+    let md5 = args.get_flag(options::MD5);
+    let sha1 = args.get_flag(options::SHA1);
+
+    let namespace = args.get_one(options::NAMESPACE);
+    let name: Option<&String> = args.get_one(options::NAME);
+
+    // https://github.com/clap-rs/clap/issues/1537
+    if !(md5 || sha1) && (namespace.is_some() || name.is_some()) {
+        return Err(USimpleError::new(
+            1,
+            "--namespace and --name arguments require either --md5 or --sha1",
+        ));
+    }
+
+    if args.get_flag(options::RANDOM) {
+        println!("{}", Uuid::new_v4());
+    } else if args.get_flag(options::TIME) {
+        let node_id = get_node_id().unwrap_or_else(|| {
+            let mut default: [u8; 6] = rand::random();
+            default[0] |= 0x01;
+            default
+        });
+        println!("{:?}", Uuid::now_v1(&node_id));
+    } else if md5 || sha1 {
+        let f = if md5 { Uuid::new_v3 } else { Uuid::new_v5 };
+
+        println!(
+            "{:?}",
+            f(
+                namespace.expect("namespace arg to be set"),
+                name.expect("name to be set").as_bytes()
+            )
+        );
+    }
+
+    Ok(())
+}
+
+pub fn uu_app() -> Command {
+    let all_uuid_types = [options::RANDOM, options::TIME, options::MD5, options::SHA1];
+
+    Command::new(uucore::util_name())
+        .version(crate_version!())
+        .about(ABOUT)
+        .override_usage(format_usage(USAGE))
+        .infer_long_args(true)
+        .arg_required_else_help(true)
```

Comment:
I think this is incorrect because the original `uuidgen` doesn't require any arguments.